app/scripts/controllers/transactions/tx-gas-utils.js

```diff
@@ -30,14 +30,10 @@ class TxGasUtil {
     try {
       estimatedGasHex = await this.estimateTxGas(txMeta, block.gasLimit)
     } catch (err) {
-      const simulationFailed = (
-        err.message.includes('Transaction execution error.') ||
-        err.message.includes('gas required exceeds allowance or always failing transaction')
-      )
-      if (simulationFailed) {
-        txMeta.simulationFails = true
-        return txMeta
+      txMeta.simulationFails = {
```

Comment:
i would consider keeping the same key name `simulationFailed` unless their is also the ui changes to go with this

Reply:
oooh i see now